### DISCUSSION TITLE
Static 404 errors in different languages.

### DISCUSSION THREAD

**Discussion by chiqui3d** [vercel/next.js#11127] [Help]
Although I see well that you force to always generate 404.html statically, sometimes as in my case I need it to be dynamic because the content depends on the language, do you know any way to do it easily?

Another thing, I have realized that it is generating two `404.html` files, one in the `out `root and another in the `out/404/404.html` folder, do you know what is due?

**Comment by yaohuangguan** [comment]
Hi, you can use a  _error.js file in your pages folder, and depends on how you set the language env variables, change the language accordingly.

**Reply by timneutkens** [reply]
Using _error is discouraged for 404.

**Comment by timneutkens** [comment]
If you need dynamic content you can fetch the dynamic content client-side.

**Comment by wereHamster** [comment]
My approach to this will likely be to render the 404 page (pages/404.js) multiple times with different query params, and place them into different locations during static export (`/en/404.html`, `/de/404.html`, …)
